src/main/java/com/frameworkium/core/ui/pages/Visibility.java

```diff
@@ -58,13 +61,36 @@ public Visibility(Wait<WebDriver> wait, JavascriptExecutor driver) {
      */
     public void waitForAnnotatedElementVisibility(Object pageObject) {
 
-        Field[] allFields = pageObject.getClass().getDeclaredFields();
-        Arrays.stream(allFields)
+        Class <? extends Object> clazz = pageObject.getClass();
+
+        // Get the declared fields from the current class
+        final List<Field> allFields = new ArrayList<Field>(Arrays.asList(clazz.getDeclaredFields()));
+
+        // Get the declared fields from the super class
+        final List<Field> superClassFields = getDeclaredFieldsFromSuperClasses(clazz);
+
+        Stream.concat(allFields.stream(),superClassFields.stream())
                 .filter(this::validateFieldVisibilityAnnotations)
                 .forEach(field ->
                         invokeWaitFunctionForField(field, pageObject));
     }
 
+    private List<Field> getDeclaredFieldsFromSuperClasses(Class <? extends Object> clazz) {
+        final List<Field> fields = new ArrayList<Field>();
+
+        // Get any declared fields from super classes
+        // i.e. when a page object extends a custom class which itself extends HtmlElement
+        for(clazz = clazz.getSuperclass();
```

Comment:
Create a new variable e.g. `Class<?> c = clazz.getSuperclass();` rather than reassign the method parameter.